depin/services/roles/xai/tasks/commands/install.yml

```diff
@@ -0,0 +1,64 @@
+---
+- name: Create program folder
+  become: true
+  ansible.builtin.file:
+    state: directory
+    owner: root
+    group: root
+    mode: '0755'
+    dest: /opt/xai/{{ xai_version }}
+
+
+- name: Download binary to tmp
+  become: true
+  ansible.builtin.unarchive:
+    remote_src: true
+    src: "{{ xai_download_url }}"
+    dest: /tmp
+  failed_when: false
+
+- name: Find binary
+  become: true
+  ansible.builtin.find:
+    paths:
+      - /tmp
+      - "{{ xai_download_path | default('/tmp') }}"
+    patterns: 'sentry-node-cli-linux$'
+    use_regex: true
+    get_checksum: true
+    recurse: true
+  register: _download
+
+- name: Install node
+  when: _download['files'] | length > 0
+  become: true
+  vars:
+    _files: |
+      {{ 
+        _download['files'] |
+        selectattr('checksum', 'equalto', xai_checksum | default(''))
+      }}
+    _path: "{{ (_files | first | default(_download['files'] | first))['path'] }}"
+  block:
+    - name: Copy required files
+      ansible.builtin.copy:
+        remote_src: true
+        src: "{{ _path }}"
+        dest: /opt/xai/{{ xai_version }}
+        mode: '0755'
+
+
+    - name: Symlink binary to path
+      become: true
+      ansible.builtin.file:
+        src: /opt/xai/{{ xai_version }}/{{ _path | basename }}
+        path: /usr/local/bin/xai
+        mode: '0755'
+        owner: root
+        group: root
+        state: hard
+        force: true        
```

Comment:
All of this can be replaced with the following as long as the `defaults/main.yml` has been updated with `xai_version, xai_download_urls, and xai_programs`

```yml
- name: Install programs
  ansible.builtin.include_role:
    name: depin.core.installer
```

Reply:
Yes we can do it after receiving license key and test wallet - for right now impossible to correctly install and run node